frontend/src/pages/InvitePage/InvitePage.tsx

```diff
@@ -0,0 +1,169 @@
+import React, { useGetIdentity, useNotify, useTranslate } from 'react-admin';
+import { useNavigate } from 'react-router-dom';
+import { useEffect, useState } from 'react';
+import { useOutbox } from '@semapps/activitypub-components';
+import { Box, CircularProgress } from '@mui/material';
+import InvitePageViewLoggedOut from './InvitePageViewLoggedOut';
+import InvitePageViewLoggedIn from './InvitePageViewLoggedIn';
+import InvitePageSuccess from './InvitePageSuccess';
+import ChoosePodProviderPage from './InvitePageProviderSelect';
+import SimpleBox from '../../layout/SimpleBox';
+
+/** The URI is expected to be encoded in the URI fragment in the SearchParams format. */
+const getCapabilityUri = (location: Location) => {
+  // const searchFragment = location.toString().match('#(.*)')?.[1];
+  // if (!searchFragment) {
+  //   return undefined;
+  // }
+
+  try {
+    const { searchParams } = new URL(location.toString());
+    // const searchParams = new URLSearchParams(searchFragment);
+    return searchParams.get('capabilityUri') || undefined;
+  } catch (e) {
+    return undefined;
+  }
+};
+
+/**
+ * Renders the main component that navigates through the invite workflow.
+ */
+const InvitePage = () => {
+  const navigate = useNavigate();
+  const translate = useTranslate();
+  const capabilityUri = getCapabilityUri(window.location)!;
+  const [showSuccess, setShowSuccess] = useState(false);
+  const [showProviderSelect, setShowProviderSelect] = useState(false as false | 'login' | 'signup');
+  const [inviterProfile, setInviterProfile] = useState(null as null | Record<string, unknown>);
+  const [capability, setCapability] = useState(null as null | Record<string, unknown>);
+  const notify = useNotify();
+  const { data: identity } = useGetIdentity();
+  const ownProfile = identity?.profileData;
+
+  const outbox = useOutbox();
+
+  if (!capabilityUri) {
+    notify('app.notification.invite_cap_invalid');
+    navigate('/');
+  }
+
+  // Fetch the capability document of the invitation.
+  useEffect(() => {
+    fetch(capabilityUri, { headers: { Accept: 'application/ld+json' } })
+      .then(async r => (r.ok ? r : Promise.reject(r.statusText)))
+      .then(async r => r.json())
```

Comment:
With the data provider correctly configured with a `Capability` resource, we could avoid the code duplication also here.